packages/terra-pills/src/FilterPills.jsx

```diff
@@ -0,0 +1,448 @@
+import React, {
+  useCallback, useLayoutEffect, useRef, useState,
+} from 'react';
+import {
+  KEY_BACK_SPACE,
+  KEY_DELETE,
+  KEY_END,
+  KEY_HOME,
+  KEY_LEFT,
+  KEY_RIGHT,
+} from 'keycode-js';
+import classNames from 'classnames';
+import classNamesBind from 'classnames/bind';
+import PropTypes from 'prop-types';
+import ResizeObserver from 'resize-observer-polyfill';
+import ThemeContext from 'terra-theme-context';
+import { injectIntl } from 'react-intl';
+import VisuallyHiddenText from 'terra-visually-hidden-text';
+import Pill from './private/_Pill';
+import PillsUtils from './private/_PillsUtils';
+import RollUpPill from './private/_RollupPill';
+import styles from './private/Pill.module.scss';
+
+const cx = classNamesBind.bind(styles);
+
+const propTypes = {
+  /**
+   * The string that labels the collection of pills, used in cases where the text label is not visible on
+   * the screen and required for minimum accessibility standards. Providing this prop adds the `aria-label`
+   * attribute to the Pill List container element. (Required)
+   */
+  ariaLabel: PropTypes.string.isRequired,
+  /**
+   * If a visible text label is used with the collection of pills, provide a string of the ID for the html
+   * element containing the visible text label. Supplying the 'ariaLabelledBy' prop will override the 'ariaLabel'
+   * prop and adds the `aria-labelledby` attribute instead to the Pill List container element.
+   *
+   * ![IMPORTANT](https://badgen.net/badge/UX/Accessibility/blue) For best practices, ensure the visible text
+   * in the html element `id` provided to the Pill List `ariaLabelledby` prop matches the text provided to the
+   * `ariaLabel` prop, for consistency in the case of fallback or errors.
+   */
+  ariaLabelledBy: PropTypes.string,
+  /**
+   * If additional visible information text is used, provide a string containing the IDs for html elements that
+   * help describe the intent of the group of pills.
+   */
+  ariaDescribedBy: PropTypes.string,
+  /**
+   * The content to be shown in the pills container. Should only contain the sub-component `<Pills.Pill>`.
+   */
+  children: PropTypes.node,
+  /**
+   * Indicates if the Pills container is rolled up or not.
+   */
+  isCollapsible: PropTypes.bool,
+  /**
+   * Callback function to remove a pill, returns 'pillKey' and 'metadata' props from the removed `<Pills.Pill>` sub-component.
+   */
+  onRemove: PropTypes.func,
+  /**
+   * @private
+   * The intl object to be injected for translations.
+   */
+  intl: PropTypes.shape({ formatMessage: PropTypes.func }).isRequired,
+};
+
+const defaultProps = {
+  ariaLabelledBy: undefined,
```

Comment:
Setting default props to `undefined` is unnecessary. An unset prop will have the value of `undefined` anyway.

Reply:
changed cfb6aad5b38a9957f1449aa2f75bbf5d001c3c96